src/algo.rs

```diff
@@ -184,6 +184,45 @@ pub fn scc<N, E, Ty, Ix>(g: &Graph<N, E, Ty, Ix>) -> Vec<Vec<NodeIndex<Ix>>>
     sccs
 }
 
+/// Condense every strongly connected component into a single node and return the result.
+///
+/// If `make_acyclic` is true, self-loops and multi edges are ignored, guaranteeing that
+/// the output is acyclic.
+pub fn condensation<N, E, Ty, Ix>(g: Graph<N, E, Ty, Ix>, make_acyclic: bool) -> Graph<Vec<N>, E, Ty, Ix>
```

Comment:
I just noticed a small issue with the signature (slightly too late, I guess :smile:): Ignoring self-loops and multi edges are really two orthogonal tasks. In particular, multi edges do not affect acyclicity, of course.

I mainly introduced the parameter because, in my use case, I have unweighted edges and am not interested in self-loops or multi edges. On the other hand, I also wanted to provide a way to preserve all weights from the input graph.